tests/e2e/pageObjects/Viewer.ts

```diff
@@ -0,0 +1,59 @@
+export class Viewer {
+  elements: Readonly<Record<string, string>> = {
+    appTopBar: '.oc-app-top-bar .oc-resource',
+    appTopBarResourceBasename: '.oc-resource-basename',
+    appTopBarResourceExtension: '.oc-resource-extension',
+    modelViewport: '#preview .model-viewport',
+    modelViewportWrapper: '#preview #scene-wrapper',
+    modelViewportWrapperFullscreen: '#scene-wrapper:fullscreen',
+    modelViewportDescription: '#preview h1.oc-invisible-sr',
+    modelViewportCanvas: '#preview .model-viewport canvas',
+    controlButtonPrev: '.preview-controls-previous',
+    controlButtonNext: '.preview-controls-next',
+    controlButtonFullscreen: '.preview-controls-fullscreen'
+  }
+
+  async getViewportDescription(): Promise<string> {
+    return await global.page.locator(this.elements.modelViewportDescription).innerText()
+  }
+
+  async getTopbarResourceName(): Promise<string> {
+    const topbarResourceBasename = await global.page
+      .locator(this.elements.appTopBarResourceBasename)
+      .innerText()
+    const topbarResourceExtension = await global.page
+      .locator(this.elements.appTopBarResourceExtension)
+      .innerText()
+    return topbarResourceBasename + topbarResourceExtension
+  }
+
+  async getViewportWrapperSize(): Promise<[string, string]> {
+    const element = await global.page.waitForSelector(this.elements.modelViewportWrapper)
+    const viewportHeight = await element.evaluate((el) => {
+      return window.getComputedStyle(el).getPropertyValue('height')
+    })
+    const viewportWidth = await element.evaluate((el) => {
+      return window.getComputedStyle(el).getPropertyValue('width')
+    })
+    // slice removes 'px' at the end of value
+    return [viewportHeight.slice(0, -2), viewportWidth.slice(0, -2)]
+  }
+
+  async getWindowInnerSize(): Promise<[string, string]> {
+    const windowInnerHeight = await global.page.evaluate(() => window.innerHeight)
+    const windowInnerWidth = await global.page.evaluate(() => window.innerWidth)
+    return [windowInnerHeight.toString(), windowInnerWidth.toString()]
+  }
+
+  async toggleFullscreenMode(): Promise<void> {
```

Comment:
```suggestion
  async toggleFullScreenMode(): Promise<void> {
```
follow same naming convention for whole file

Reply:
@amrita-shrestha since the same method is called for turning fullscreen mode on and off. Any suggestions for naming?

Reply:
its about convention `camelCase` you typed `s` rather than `S`

Reply:
Ok, I get your point now. I guess in this case we better write "full screen" instead of "fullscreen" to be consistent...